frontend/javascripts/oxalis/view/input_catcher.js

```diff
@@ -48,7 +49,13 @@ function adaptInputCatcher(inputCatcherDOM: HTMLElement, makeQuadratic: boolean)
 const renderedInputCatchers = new Map();
 
 export function recalculateInputCatcherSizes() {
-  const viewportRects = {};
+  const viewportRects: Object = {
```

Comment:
This is a bit of a dirty workaround. I'm not sure yet which component cannot deal with the TD viewport rect, but this at least fixes the problem. @MichaelBuessemeyer Do you know how this could be turned into a proper fix?

Reply:
Instead of falling back to the "default viewport rect", I now changed it to the "empty viewport rect". It's still not super elegant, but:

- I find it better to have a key for each viewport (even if it contains a zero-area-rect). I think, most of the other code assumes that they can look up each plane's viewport rect.
- it fixes another issue I had. Merely moving the `if (!this.controls)`-check in the `td_controller`, still caused a small rendering bug for me on initial load. See screenshot:
![image](https://user-images.githubusercontent.com/2486553/116400352-3bc82f80-a82a-11eb-9ee3-908c371c6662.png)

Reply:
> I find it better to have a key for each viewport (even if it contains a zero-area-rect). I think, most of the other code assumes that they can look up each plane's viewport rect.

I do not think that this should be an issue as the default state has a viewport rect for each viewport and the `setInputCatcherRects`-Action only updates the rects of the passed viewports. Therefore there should always be a rect for each viewport.

>it fixes another issue I had. Merely moving the if (!this.controls)-check in the td_controller, still caused a small rendering bug for me on initial load. See screenshot:

:+1: 

Additionally, I think having "empty" viewports as default should also solve @fm3 bug as my guess is that this is caused by having a non-empty viewport for a viewport that is not rendered in the dom.
![image](https://user-images.githubusercontent.com/39529669/116548144-19024d80-a8f4-11eb-9f6d-9f226cca2188.png)

Reply:
> I do not think that this should be an issue as the default state has a viewport rect for each viewport and the setInputCatcherRects-Action only updates the rects of the passed viewports. Therefore there should always be a rect for each viewport.

Ah, that explains why it didn't crash. However, as you indicated, not updating not-rendered viewports leads to stale values. So, I think my changes should stay as is.

Reply:
>So, I think my changes should stay as is.

Indeed think so too. (But forgot to make this explicit :grin:)

Reply:
Then only an explicit GitHub "Accept" is missing, I think :)